src/gwt/src/org/rstudio/studio/client/workbench/views/source/model/DocUpdateSentinel.java

```diff
@@ -722,23 +722,37 @@ private void applyProperties(JsObject properties,
       }
    }
    
-   public void suspendAutosave(boolean value)
+   public void withChangeDetectionSuspended(Command command)
    {
-      suspendAutosave_ = value;
+      try
+      {
+         suspendDetectChanges_ += 1;
```

Comment:
Note: using an integer flag here just to make this API safe against re-entrant calls to `withChangeDetectionSuspended()`.